packages/abi-gen/test-cli/expected-output/python/abi_gen_dummy.py

```diff
@@ -11,6 +19,30 @@
 from zero_ex.contract_wrappers.tx_params import TxParams
 
 
+class InnerStructDescription(  # noqa: E501 (line too long)
+    TypedDict
+):
+    """Python representation of a tuple or struct."""
+
+    inner_struct: SomeBytesAnIntegerADynamicArrayOfBytesAString  # noqa: E501 (line too long)
+
+    description: str
+
+class SomeBytesAnIntegerADynamicArrayOfBytesAString(  # noqa: E501 (line too long)
```

Comment:
Instead of concatenating all the properties as the name here, let's use `typedDict0`, `typedDict1`, etc... which is also what `go-ethereum's` `abigen` tool does. Is it possible to then make named classes in a separate file that inhert from these classes? This way, users will see and be able to use the properly naming classes that we will manually maintain, but if we ever forget to create an alias, it'll default to `typedDictX`?

This is what we do in Mesh: https://github.com/0xProject/0x-mesh/blob/development/ethereum/wrappers/order_validation_utils_aliases.go

Reply:
The problem is that some sort of state would need to be held, such that when the references to structs are generated they can refer back to the the classes that have already been declared.  This could probably be addressed with a state variable in `abi-gen/index.src` that's captured in the closure for the helpers, but... There's also the complication of, when generating the class declarations, being able to support nested definitions, which would require multiple passes over all of the methods, since references to struct A within struct B depend on having already encountered struct A, which may not always be the case.  There might be a relatively elegant solution to this, but IMO it's overly convoluted,  and the code will be hard to read, review and maintain, at least as compared to the idea of just using a deterministic way of generating the `TypedDict` names.

For the time being, I'd prefer to leave the deterministic name generation method in place, and only bring in the state convolution later, and only if necessary.  If you'd _really_ like to see this happen, I'd vote for leaving it out of this PR and creating a follow-on Issue to track it.

However, I agree that the way it's being done now, with these super long names, is quite ugly, but I chose to do it that way anyways, because it was easy and it works.  Thinking further, though, what do you think about using a nicer but still deterministic approach?  I'm envisioning `TypedDict9fdb6c`, where it's just `TypedDict` concatenated with a truncated hash of the method names.  This would provide a short but unique name which could be determined anywhere (declarations, references, nested declarations, etc) without needing to refer back to any name choices made earlier in the generation process.

Reply:
I didn't address your comment about type aliases.  I'm confused about how they fit in to the generated code.  Do you expect they'll appear in the generated code at all?  Or are they just for the user of the wrapper to use in constructing arguments and deconstructing return values?

Reply:
I went ahead with my proposal of naming type classes with the hash of their components' names.  I introduced type aliases to the same Python module that contains the Exchange wrapper, seen [here](https://github.com/0xProject/0x-monorepo/blob/51d9b45ec44ca14616f4ac6c3f35c1b999275197/python-packages/contract_wrappers/src/zero_ex/contract_wrappers/exchange/types.py#L21-L65).  Unfortunately the Exchange API itself still uses weird names (eg `Tuple0xffffff`) for its parameter specification, but the client can write calls in terms of the type aliases, as shown in the module docstring [here](https://github.com/0xProject/0x-monorepo/blob/51d9b45ec44ca14616f4ac6c3f35c1b999275197/python-packages/contract_wrappers/src/zero_ex/contract_wrappers/__init__.py#L120-L174).

Reply:
Re: your second paragraph question -- it's only for users to construct the arguments and deconstructing return values.

Reply:
I'm happy with this solution! Thanks